src/Microsoft.DotNet.Build.Tasks/PackageFiles/codeOptimization.targets

```diff
@@ -0,0 +1,81 @@
+          Condition="'$(OSEnvironment)'=='Windows_NT' and '$(EnableProfileGuidedOptimization)'=='true' and Exists('$(OptimizationDataDir)/$(AssemblyName).zip')">
+
+    <PropertyGroup>
+      <IBCMergeToolPath>$(ToolsDir)/ibcmerge.exe</IBCMergeToolPath>
+      <OptimizationDataForAssemblyDir>$(IntermediateOutputPath)/OptimizationData/$(AssemblyName)</OptimizationDataForAssemblyDir>
+      <OptimizedAssemblyFile>$(IntermediateOutputPath)/OptimizedAssembly/$(AssemblyName).dll</OptimizedAssemblyFile>
+    </PropertyGroup>
+
+    <!-- Extract the profile data from the archive -->
+    <ExtractZipFile SourceArchive="$(OptimizationDataDir)/$(AssemblyName).zip"
+                    DestinationDirectory="$(OptimizationDataForAssemblyDir)"
+                    OverwriteDestination="true" />
+
+    <!-- Copy the compiled assembly into a folder for further processing -->
+    <MakeDir Directories="$(IntermediateOutputPath)/OptimizedAssembly" />
+    <Message Text="Copy @(IntermediateAssembly) to $(IntermediateOutputPath)/OptimizedAssembly" />
+    <Copy SourceFiles="@(IntermediateAssembly)" DestinationFolder="$(IntermediateOutputPath)/OptimizedAssembly" />
+
+    <!-- Delete existing optimization data from the file that came from with the training data -->
+    <Exec Command="$(IBCMergeToolPath) -f -delete -mo $(OptimizationDataForAssemblyDir)/$(AssemblyName).dll $(OptimizationDataForAssemblyDir)/$(AssemblyName).ibc" />
+
+    <!-- Create ibc data for the new assembly -->
+    <Exec Command="$(IBCMergeToolPath) -f -o $(IntermediateOutputPath)/OptimizedAssembly/$(AssemblyName).new.ibc -delete -mo $(OptimizedAssemblyFile) -incremental $(OptimizationDataForAssemblyDir)/$(AssemblyName).dll" />
+
+    <!-- Compact the optimization data in the optimized assembly -->
+    <Exec Command="$(IBCMergeToolPath) -minify -mu $(OptimizedAssemblyFile) -partialNGEN" />
+
+    <!-- Verify that the optimization data has been applied -->
+    <Exec Command="$(IBCMergeToolPath) -mi $(OptimizedAssemblyFile)" />
+
+    <!-- We need to make sure that the assembly that gets packaged is the one with the optimization data -->
+    <ItemGroup>
+      <IntermediateAssembly Remove="@(IntermediateAssembly)"/>
+      <IntermediateAssembly Include="$(IntermediateOutputPath)/OptimizedAssembly/$(AssemblyName).dll"/>
+    </ItemGroup>
+  </Target>
+
+  <!-- We need the OptimizationData package in order to be able to optimize the assembly -->
+  <Target Name="RestoreOptimizationDataPackage" BeforeTargets="CoreCompile"
+          Condition="'$(EnableProfileGuidedOptimization)'=='true' and !Exists('$(OptimizationDataDir)\project.json')">
+
+    <!-- Dynamically create a project.json file used to restore the optimization data-->
+    <Message Text="Generating project.json for optimization data"  Importance="low" />
+    <ItemGroup>
+      <_OptimizationDataJsonLine Include="{&quot;dependencies&quot;: {" />
```

Comment:
I had it on one line, but then I moved it to separate lines where the line in the middle is the package name. That way we can easily add more packages (if we have to).